test/dotnet-vstest.Tests/VSTestTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+using System;
+using System.IO;
+using FluentAssertions;
+using Microsoft.DotNet.TestFramework;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Cli.VSTest.Tests
+{
+    public class VSTestTests: TestBase
+    {
+        [Fact]
+        public void TestsFromAGivenContainerShouldRunWithExpectedOutput()
+        {
+            // Copy DotNetCoreTestProject project in output directory of project dotnet-vstest.Tests
+            string testAppName = "DotNetCoreTestProject";
+            TestInstance testInstance = TestAssetsManager.CreateTestInstance(testAppName).WithLockFiles();
+
+            string testProjectDirectory = testInstance.TestRoot;
+
+            // Build project DotNetCoreTestProject
+            new Build3Command()
+                .WithWorkingDirectory(testProjectDirectory)
+                .Execute()
+                .Should()
+                .Pass();
+
+            // Prepare args to send vstest
+            string configuration = Environment.GetEnvironmentVariable("CONFIGURATION") ?? "Debug";
+            string testAdapterPath = Path.Combine(testProjectDirectory, "bin", configuration, "netcoreapp1.0");
+            string outputDll = Path.Combine(testAdapterPath, $"{testAppName}.dll");
+            string argsForVstest = string.Concat("\"", outputDll, "\"", " --TestAdapterPath:", "\"", testAdapterPath, "\"", " --Platform:x64");
```

Comment:
Not exactly, this is not representative or not exactly, it won't just be the dll? Is the plan that the test target will be able to put all this together and invoke the tool for me then?

So that I can just do dotnet test in a folder containing a test project in it?
Also, say, for CI, I will have to find the adapter (isn't it commonly binplaced with the app? can't you try some heuristics to find the adapter before forcing the user to find it for you?).

Why is x86 the default? I would imagine that x64 is actually the more common one these days no? And why need to specify it at all?

If this command is expected to be invoked directly from the command line, its inputs need to be simple and intuitive.